compression.go

```diff
@@ -1,6 +1,22 @@
 package kafka
 
-import "sync"
+import (
+	"io"
+	"sync"
+
+	"github.com/pkg/errors"
```

Comment:
I don't think we have a dependency on this package right now, and it doesn't look like we're using much of the features it provides.

Any reasons not to use the standard `errors` package instead?